libs/wlgen/wlgen/rta.py

```diff
@@ -505,14 +505,19 @@ def conf(self,
 
         self.loadref = loadref
 
-        # Setup class-specific configuration
-        if kind == 'custom':
-            self._confCustom()
-        elif kind == 'profile':
-            self._confProfile()
-
-        # Move configuration file to target
-        self.target.push(self.json, self.run_dir)
+        try:
+            tf = NamedTemporaryFile(delete=False)
+            # Setup class-specific configuration
+            if kind == 'custom':
+                self._confCustom(tf)
+            elif kind == 'profile':
+                self._confProfile(tf)
+
+            # Move configuration file to target
+            self.target.push(tf.name, os.path.join(self.run_dir,self.json))
+        finally:
```

Comment:
the "finally" clause becomes unnecessary as the as the context manager will close and delete the file. That also avoids the issue of tf not being defined if NamedTemporaryFile raises an exception while creating the file.